lib/vim_state.ahk

```diff
@@ -60,7 +61,16 @@
     if(LineCopy!=-1){
       this.LineCopy := LineCopy
     }
-    this.CheckMode(this.Vim.Conf["VimVerbose"]["val"], Mode, g, n, LineCopy)
+    this.UpdateModeIndicators(Mode, g, n, LineCopy)
+  }
+
+  UpdateModeIndicators(Mode, g, n, LineCopy){
```

Comment:
I would like to include this function directly in the SetMode, if `UpdateModeIndicators` is not used from other places.

Reply:
I'm not sure what you mean by "Include this function directly in the SetMode"

Reply:
If there is a test or necessities to call it alone, then it is needed.
Otherwise it is not necessary in such a short function.

At least `ShowModeInfo` (`CheckMode`) must be in SetMode, as it is not special for the indicator update.

Reply:
I strongly disagree with your first point. It is a fundamental principle of readable code to have them in a separate function. Most of the time, anyone skimming `setMode` does not have to know how the mode indicators are updated, only that `SetMode` does update them.
I highly recommend Robert Martin's book, [Clean Code](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882) for an explanation of the principles behind it, if I haven't misinterpreted your argument.

I don't understand why you don't think that `ShowModeInfo` is not part of the indicator. Is this because you only use `ShowModeInfo` for debugging? But even then, the purpose is to indicate the mode (at least, at `verbose=1` it is)

If you insist that `ShowModeInfo` should not be in a separate function (even a renamed function, if you think `UpdateModeIndiators` is the wrong name for it), then it may be reasonable to inline (move back to `SetMode`) this function.

Reply:
Some people says too much encapsulations make it difficult to read. 
This is a kind of philosophy, and it is meaningless to discuss more.

Reply:
Sure, you're the boss. I'll revert.